src/main/java/org/opensearch/neuralsearch/query/HybridQuery.java

```diff
@@ -46,7 +46,8 @@ public HybridQuery(final Collection<Query> subQueries, final List<Query> filterQ
         if (subQueries.isEmpty()) {
             throw new IllegalArgumentException("collection of queries must not be empty");
         }
-        if (hybridQueryContext.getPaginationDepth() == 0) {
+        Integer paginationDepth = hybridQueryContext.getPaginationDepth();
+        if (paginationDepth != null && paginationDepth == 0) {
```

Comment:
```suggestion
        if (Objects.nonNull(paginationDepth) && paginationDepth == 0) {
```
this way it's consistent with another similar check in this PR